lib/logflare/backends/adaptor/postgres_adaptor/shared_repo.ex

```diff
@@ -0,0 +1,97 @@
+defmodule Logflare.Backends.Adaptor.PostgresAdaptor.SharedRepo do
+  use Ecto.Repo,
+    otp_app: :logflare,
+    adapter: Ecto.Adapters.Postgres
+
+  require Logger
+
+  alias Logflare.Source
+  alias Logflare.Backends.Backend
+
+  alias Logflare.Backends.Adaptor.PostgresAdaptor.Supervisor
+  alias Logflare.Backends.Adaptor.PostgresAdaptor.Repo.Migrations
+
+  def start(%Backend{} = backend) do
+    config = backend.config
+    pool_size = Application.get_env(:logflare, :postgres_backend_adapter)[:pool_size] || 10
```

Comment:
lets set default pool size in config.exs instead, and default needs to be small from filipe's prior testing around the pool sizes
see https://supabase.slack.com/archives/C032VRDHPDE/p1688281725600829?thread_ts=1688146906.543309&cid=C032VRDHPDE